element_deeplabcut/dlc.py

```diff
@@ -251,28 +275,44 @@ def make(self, key):
 @schema
 class BodyPart(dj.Lookup):
     definition = """
-    body_part: varchar(32)
+    body_part                : varchar(32)
     ---
-    body_part_description='': varchar(1000)
+    body_part_description='' : varchar(1000)
     """
 
     @classmethod
-    def insert_from_config(cls, dlc_config: dict):
+    def insert_from_config(cls, dlc_config: dict, descriptions: list = None):
+        """Insert all body parts from a config file
+
+        :param dlc_config: path to a config.y*ml, or dict including contents thereof
+        :param descriptions: optional list describing new body parts
+        """
+
         # handle dlc_config being a yaml file
         if not isinstance(dlc_config, dict):
-            dlc_config_fp = pathlib.Path(dlc_config)
-            if dlc_config_fp.exists() and dlc_config_fp.suffix in ('.yml', '.yaml'):
-                with open(dlc_config, 'rb') as f:
+            dlc_config_fp = find_full_path(get_dlc_root_data_dir(),
+                                           pathlib.Path(dlc_config))
+            assert dlc_config_fp.exists(), ('dlc_config is neither dict nor filepath'
+                                            + f'\n Check: {dlc_config_fp}')
+            if dlc_config_fp.suffix in ('.yml', '.yaml'):
+                with open(dlc_config_fp, 'rb') as f:
                     dlc_config = yaml.safe_load(f)
         # -- Check and insert new BodyPart --
-        if 'bodyparts' in dlc_config:
-            tracked_body_parts = cls.fetch('body_part')
-            new_body_parts = np.setdiff1d(dlc_config['bodyparts'], tracked_body_parts)
-            if new_body_parts:
-                print(f'Existing body parts: {tracked_body_parts}')
-                print(f'New body parts: {new_body_parts}')
-                if dj.utils.user_choice(f'Insert {len(new_body_parts)} new body part(s)?') == 'yes':
-                    cls.insert({'body_part': b} for b in new_body_parts)
+        assert 'bodyparts' in dlc_config, f'Found no bodyparts section in {dlc_config}'
+        tracked_body_parts = cls.fetch('body_part')
+        new_body_parts = np.setdiff1d(dlc_config['bodyparts'], tracked_body_parts)
+        if new_body_parts is not None:  # Required bc np.array is ambiguous as bool
+            print(f'Existing body parts: {tracked_body_parts}')
+            print(f'New body parts: {new_body_parts}')
+            if descriptions:
+                assert len(descriptions)\
+                        == len(new_body_parts), ('Descriptions list does not match '
+                                                 + ' the number of new_body_parts')

```

Comment:
I think this assertion will fail most of the time. Because the `len(new_body_parts)` will not match the `len(descriptions)`
In our function, we have logic for checking which body-part from the config file are new, and only insert those new ones - the `new_body_parts`. In the logic here, we are expecting the users to know, at the time they call this function, what are the new body-parts, and in which order, so they can provide a meaningful `descriptions`.
This is a bit unrealistic and error prone, I'd suggest not even requiring `descriptions` here. The body_part description can be updated after the fact fairly easily and less error prone.

Reply:
Generally we do want to avoid updating, but for metadata at this level, updating is quite common, especially updating a description field (it would be another conversation if we're updating fields such as sampling rate for example). I would rather allow for updating than introducing more complex logic and usage requirement.